### PR TITLE
AB#89349 Added planning status of site

### PR DESCRIPTION
**Pull request by Antek012**

### CODE REVIEW

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public PlanningDetails(SitePlanningStatus planningStatus)
+    {
+        Build(planningStatus).CheckErrors();
+    }
+
+    public SitePlanningStatus PlanningStatus { get; private set; }
+
+    public bool IsAnswered()
+    {
+        return false;
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs:20]: Is it correct?

File: src/HE.Investment.AHP.WWW/Controllers/SiteController.cs [hunk 2]
```diff
@@ -343,4 +380,26 @@ protected override async Task<IStateRouting<SiteWorkflowState>> Routing(SiteWork
...
+    {
+        var siteId = this.GetSiteIdFromRoute();
+
+        return await this.ExecuteCommand<TViewModel>(
+            _mediator,
+            command,
+            async () => await ContinueWithRedirect(new { siteId }),
+            async () =>
+            {
+                var deliveryPhaseDetails = await _mediator.Send(new GetSiteQuery(siteId.Value), cancellationToken);
```
Comment (Carq) [inline src/HE.Investment.AHP.WWW/Controllers/SiteController.cs:398]: `deliveryPhaseDetails` ➡️ `siteDetails`

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace HE.Investment.AHP.Domain.Site.ValueObjects;
+
+public class PlanningDetails : ValueObject, IQuestion
+{
+    public PlanningDetails(SitePlanningStatus planningStatus)
+    {
+        Build(planningStatus).CheckErrors();
+    }
+
+    public SitePlanningStatus PlanningStatus { get; private set; }
```
Comment (Carq) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs:16]: Do you plan to put more into the Value Object? If not then I probably Value Object is not needed.

**Review by Carq — APPROVED** [review]

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs (hunk 3 above)
Comment (Antek012) [reply src/HE.Investment.AHP.Domain/Site/ValueObjects/PlanningDetails.cs:16]: yes, this object will be extended

**Pull request merged**